src/browserlib/create-outline.js

```diff
@@ -0,0 +1,352 @@
+/**
+ * Creates an outline for the DOM subtree rooted at the given sectioning content
+ * or sectioning root element.
+ *
```

Comment:
can we re-use an existing module instead? e.g. https://h5o.github.io/
if not, can we move this into a separate module? this doesn't feel reffy-esque since it's not spec-specific as far as I can tell.

Reply:
First, I wish I would have found that module earlier on, that might have saved me some coding time :) I see that I ended up with very similar code as in that module, which seems good.

Now, I don't see any easy way to re-use it because the creation of the outline is not what we're interested in. We're interested in mapping elements with IDs to their enclosing section. This mapping gets done during the generation of the outline, and this is what the createOutline function returns (along with the outline). The h5o module only creates the outline and wouldn't give us the information for non heading elements.

There may be a more direct way to get that information without creating an outline. Starting from an element with an ID, it seems doable to parse the DOM tree before it (in document order) to find the closest heading for it, taking into account sectioning rules in HTML. Given the way these sectioning rules are written in the HTML spec, it just seemed easier to follow the instructions and create the outline...

The code is indeed generic and could be moved to a separate library. But that means more work, so I wouldn't mind keeping it in Reffy for now.